api/src/test/java/org/openmrs/api/ConceptServiceTest.java

```diff
@@ -2245,9 +2245,40 @@ public void getConceptsByAnswer_shouldFindAnswersForConcept() {
 	 * @see ConceptService#getConceptsByClass(ConceptClass)
 	 */
 	@Test
-	public void getConceptsByClass_shouldNotFailDueToNoNameInSearch() {
-		ConceptService cs = Context.getConceptService();
-		cs.getConceptsByClass(cs.getConceptClass(1));
+	public void getConceptsByClass_shouldGetConceptsByClass() {
+		// setup
+		ConceptClass cc = new ConceptClass(3);
+		Concept expectedConcept1 = new Concept(3);
+		expectedConcept1.setUuid("0cbe2ed3-cd5f-4f46-9459-26127c9265ab");
+		Concept expectedConcept2 = new Concept(60);
+		expectedConcept2.setUuid("35d3346a-6769-4d52-823f-b4b234bac3e3");
+		Concept expectedConcept3 = new Concept(88);
+		expectedConcept3.setUuid("15f83cd6-64e9-4e06-a5f9-364d3b14a43d");
+		Concept expectedConcept4 = new Concept(792);
+		expectedConcept4.setUuid("d144d24f-6913-4b63-9660-a9108c2bebef");
+
+		List<Concept> expectedConcepts = Arrays.asList(expectedConcept1, expectedConcept2, expectedConcept3, expectedConcept4);
+		
+		// replay
+		List<Concept> actualConcepts = conceptService.getConceptsByClass(cc);
+		
+		// verify
+		Assert.assertThat(actualConcepts, is(expectedConcepts));
+	}
+	
+	/**
+	 * @see ConceptService#getConceptsByClass(ConceptClass)
+	 */
+	@Test
+	public void getConceptsByClass_shouldReturnAnEpmtyListIfNoneWasFound() {
+		// setup
+		ConceptClass cc = new ConceptClass(23);
+		
+		// replay
+		List<Concept> concepts = conceptService.getConceptsByClass(cc);
+		
+		// verify
+		Assert.assertTrue(concepts.isEmpty());
 	}
```

Comment:
Please do this:
```java
/**
 * @see ConceptService#getConceptsByClass(ConceptClass)
 */
@Test
public void getConceptsByClass_shouldGetUnorderedConceptsForClass() {
  // replay
  List<Concept> actualConcepts = conceptService.getConceptsByClass(new ConceptClass(3));
  
  // verify
  Assert.assertThat(actualConcepts.size(), is(4));
  Assert.assertThat(actualConcepts, containsInAnyOrder(conceptService.getConcept(3), conceptService.getConcept(60), conceptService.getConcept(88), conceptService.getConcept(792)));
}

/**
 * @see ConceptService#getConceptsByClass(ConceptClass)
 */
@Test
public void getConceptsByClass_shouldReturnEmptyListForEmptyClass() {
  // replay
  List<Concept> concepts = conceptService.getConceptsByClass(new ConceptClass(23));
  
  // verify
  Assert.assertTrue(concepts.isEmpty());
}
```
It is really not ok to decouple your test method by resetting the UUIDs _again_, even though they are defined already in **standardTestDataset.xml**.

Furthermore your test did not resist reordering the loading order in **standardTestDataset.xml**.

Finally there was a typo in the second method name: `AnEpmtyList`, I suggested new names anyway.

Reply:
Thanks for the advice.. actually this makes sense and looks way better

Reply:
@samuelmale can you apply the changes on a subsequent commit?